clippy_lints/src/inherent_impl.rs

```diff
@@ -43,7 +44,7 @@ declare_clippy_lint! {
 #[allow(clippy::module_name_repetitions)]
 #[derive(Default)]
 pub struct MultipleInherentImpl {
-    impls: DefIdMap<Span>,
+    impls: Vec<(LocalDefId, Span)>,
```

Comment:
Why not put the IndexMap right here?

Reply:
Lifetime issues. Lints need a `'static` lifetime, but the `IndexMap` uses `Ty<'_>` as the key. Technically those could be cast from a pointer to a usize.

Reply:
You could remove `Span`s by using `hir.span(hir.local_def_id_to_hir_id(impl_did))`.

Reply:
> Lifetime issues.

Ah ok.

Reply:
Any interest in removing the Spans? If not I'm fine merging as is.

Reply:
Is there really much improvement from doing that? It's just changing the small amount of memory use for having to lookup the span later in the HIR map.